### ISSUE TITLE
Parallel lifecylce methods

### ISSUE THREAD

**Issue by jbadeau** [cujojs/wire#141]
Hi Brian,

We have split up our wire specs into something like:

settings.js (loads settings from servere once)
services.js (loads navigation data, conditions, links, etc..)
app.js (App, controllers, router, etc...)

In the services.js we have services with async/pomised init methods that load data from the server. we have seen that even if there is no interdependency between components that the service calls are executed in sequence rather than paralllel. This has a very big impact for us when latency is high. In our case getting data from China in Europe. We don't use lazy loading as that would also suffer from sequential loading.

Basically I think if there are no dependencies between componenets in a context than any lifecycle method that returns a promise should be resolved in a context scoped parallel promise.

Is this clear?
